### PR TITLE
config: add OpenStack-related config to the v5.4 SOS kernel

### PR DESCRIPTION
**Pull request by peterfang**
Add kernel config required to bring up OpenStack in the v5.4 SOS kernel.

Signed-off-by: Peter Fang <peter.fang@intel.com>

### CODE REVIEW

File: recipes-kernel/linux/files/sos_5.4.cfg [hunk 1]
```diff
@@ -3,3 +3,43 @@ CONFIG_ACRN_GUEST=y
...
+CONFIG_IP6_NF_TARGET_REJECT=y
+CONFIG_IP6_NF_MANGLE=y
+CONFIG_IP6_NF_RAW=y
+CONFIG_IP6_NF_NAT=y
+CONFIG_IP6_NF_TARGET_MASQUERADE=y
+CONFIG_CHECKPOINT_RESTORE=y
+CONFIG_BLK_DEV_BSGLIB=y
+CONFIG_PACKET_DIAG=y
+CONFIG_UNIX_DIAG=y
+CONFIG_BRIDGE_NF_EBTABLES=y
```
Comment (cheeyanglee) [inline recipes-kernel/linux/files/sos_5.4.cfg:21]: add  cfg/net/bridge.scc to KERNEL_FEATURES should include these bridge features

File: recipes-kernel/linux/files/sos_5.4.cfg [hunk 2]
```diff
@@ -3,3 +3,43 @@ CONFIG_ACRN_GUEST=y
 CONFIG_DWXPCS=m
 CONFIG_MARVELL_PHY=m
 CONFIG_MARVELL_10G_PHY=m
+
+# OpenStack-related settings
+CONFIG_IP6_NF_IPTABLES=y
```
Comment (cheeyanglee) [inline recipes-kernel/linux/files/sos_5.4.cfg:8]: cfg/net/ip6_nf.scc should enable netfilter for ipv6

File: recipes-kernel/linux/files/sos_5.4.cfg [hunk 3]
```diff
@@ -3,3 +3,43 @@ CONFIG_ACRN_GUEST=y
...
+CONFIG_IP6_NF_MATCH_IPV6HEADER=y
+CONFIG_IP6_NF_MATCH_RPFILTER=y
+CONFIG_IP6_NF_FILTER=y
+CONFIG_IP6_NF_TARGET_REJECT=y
+CONFIG_IP6_NF_MANGLE=y
+CONFIG_IP6_NF_RAW=y
+CONFIG_IP6_NF_NAT=y
+CONFIG_IP6_NF_TARGET_MASQUERADE=y
+CONFIG_CHECKPOINT_RESTORE=y
```
Comment (cheeyanglee) [inline recipes-kernel/linux/files/sos_5.4.cfg:17]: add features/criu/criu-enable.scc for checkpoint restore

File: recipes-kernel/linux/files/sos_5.4.cfg [hunk 4]
```diff
@@ -3,3 +3,43 @@ CONFIG_ACRN_GUEST=y
...
+CONFIG_BRIDGE_EBT_ARPREPLY=m
+CONFIG_BRIDGE_EBT_DNAT=m
+CONFIG_BRIDGE_EBT_MARK_T=m
+CONFIG_BRIDGE_EBT_REDIRECT=m
+CONFIG_BRIDGE_EBT_SNAT=m
+CONFIG_BRIDGE_EBT_LOG=m
+CONFIG_BRIDGE_EBT_NFLOG=m
+CONFIG_NETLINK_DIAG=y
+CONFIG_SCSI_ISCSI_ATTRS=m
+CONFIG_ISCSI_TCP=m
```
Comment (cheeyanglee) [inline recipes-kernel/linux/files/sos_5.4.cfg:44]: cgl/cfg/iscsi.scc for iSCSI Initiator over TCP/IP

File: recipes-kernel/linux/files/sos_5.4.cfg [hunk 5]
```diff
@@ -3,3 +3,43 @@ CONFIG_ACRN_GUEST=y
...
+CONFIG_BRIDGE_EBT_PKTTYPE=m
+CONFIG_BRIDGE_EBT_STP=m
+CONFIG_BRIDGE_EBT_VLAN=m
+CONFIG_BRIDGE_EBT_ARPREPLY=m
+CONFIG_BRIDGE_EBT_DNAT=m
+CONFIG_BRIDGE_EBT_MARK_T=m
+CONFIG_BRIDGE_EBT_REDIRECT=m
+CONFIG_BRIDGE_EBT_SNAT=m
+CONFIG_BRIDGE_EBT_LOG=m
+CONFIG_BRIDGE_EBT_NFLOG=m
```
Comment (saininav) [inline recipes-kernel/linux/files/sos_5.4.cfg:41]: Above bridge configuration is already enabled. 
http://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/tree/cfg/net/bridge.cfg

Please verify at runtime.

File: recipes-kernel/linux/files/sos_5.4.cfg [hunk 6]
```diff
@@ -3,3 +3,43 @@ CONFIG_ACRN_GUEST=y
...
+# OpenStack-related settings
+CONFIG_IP6_NF_IPTABLES=y
+CONFIG_IP6_NF_MATCH_IPV6HEADER=y
+CONFIG_IP6_NF_MATCH_RPFILTER=y
+CONFIG_IP6_NF_FILTER=y
+CONFIG_IP6_NF_TARGET_REJECT=y
+CONFIG_IP6_NF_MANGLE=y
+CONFIG_IP6_NF_RAW=y
+CONFIG_IP6_NF_NAT=y
+CONFIG_IP6_NF_TARGET_MASQUERADE=y
```
Comment (saininav) [inline recipes-kernel/linux/files/sos_5.4.cfg:16]: IP6 configuration is already enabled. Please verify at runtime.
http://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/tree/cfg/net/ip6_nf.cfg

Could you update your patch with missing configurations only. Please use *.scc files from yocto-kernel-cache, if configuration is available.

File: recipes-kernel/linux/files/sos_5.4.cfg (hunk 5 above)
Comment (wenlingz) [reply recipes-kernel/linux/files/sos_5.4.cfg:41]: YES, it's already enabled, @peterfang please drop it, thanks naveen

File: recipes-kernel/linux/files/sos_5.4.cfg (hunk 6 above)
Comment (wenlingz) [reply recipes-kernel/linux/files/sos_5.4.cfg:16]: @saininav  
CONFIG_IP6_NF_NAT=y
CONFIG_IP6_NF_TARGET_MASQUERADE=y, these two configs need to be added, @peterfang, please modify, thanks

File: recipes-kernel/linux/linux-intel-acrn-sos_5.4.bb [hunk 7]
```diff
@@ -6,4 +6,9 @@ LINUX_VERSION_EXTENSION = "-linux-intel-acrn-sos"
 
 SUMMARY = "Linux Kernel with ACRN enabled (SOS)"
 
-KERNEL_FEATURES_append = " sos_5.4.scc "
+KERNEL_FEATURES_append = "cfg/net/bridge.scc \
```
Comment (saininav) [inline recipes-kernel/linux/linux-intel-acrn-sos_5.4.bb:9]: bridge.scc is already enabled by default. It is not required to add here.

File: recipes-kernel/linux/linux-intel-acrn-sos_5.4.bb [hunk 8]
```diff
@@ -6,4 +6,9 @@ LINUX_VERSION_EXTENSION = "-linux-intel-acrn-sos"
...
+                          cfg/net/ip6_nf.scc \
```
Comment (saininav) [inline recipes-kernel/linux/linux-intel-acrn-sos_5.4.bb:10]: ip6_nf.scc is already enabled by default it is not required to add here.

**Comment by cheeyanglee** [comment]
merge+1

**Pull request merged**